src/systems/triggered_publisher/TriggeredPublisher.cc

```diff
@@ -598,31 +612,73 @@ void TriggeredPublisher::Configure(const Entity &,
   else
   {
     ignerr << "No ouptut specified" << std::endl;
-    return;
+  }
+
+  if (sdfClone->HasElement("service"))
+  {
+    for (auto serviceElem = sdfClone->GetElement("service"); serviceElem;
+         serviceElem = serviceElem->GetNextElement("service"))
+    {
+      ServiceOutputInfo serviceInfo;
+      serviceInfo.servName = serviceElem->Get<std::string>("name");
+      if (serviceInfo.servName.empty())
+      {
+        ignerr << "Service name cannot be empty\n";
+        return;
+      }
+      serviceInfo.reqType = serviceElem->Get<std::string>("reqType");
+      if (serviceInfo.reqType.empty())
+      {
+        ignwarn << "Service request type is empty\n";
+      }
+      serviceInfo.repType = serviceElem->Get<std::string>("repType");
+      if (serviceInfo.repType.empty())
+      {
+        ignwarn << "Service reply type is empty\n";
+      }
+      serviceInfo.reqMsg = serviceElem->Get<std::string>("reqMsg");
+      if (serviceInfo.reqMsg.empty())
+      {
+        ignwarn << "Service request string is empty\n";
+      }
+      this->serviceOutputInfo.push_back(std::move(serviceInfo));
+    }
+  }
+  else
+  {
+    ignwarn << "No service specified" << std::endl;
   }
```

Comment:
This whole block should probably be removed; after all, it is not a warning to have no service specified.

That said, we probably *do* want to fail if neither an `output` nor a `service` have been specified.  So it is worthwhile having a flag or something which checks to make sure at least one of them is there.